### PR TITLE
01 node basics

### PR DESCRIPTION
**Pull request by ekaterina8aleksa**

### CODE REVIEW

File: contacts.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
+const fs = require("fs").promises;
+const path = require("path");
+
+class Contacts {
+    constructor() {
+        this.fileContactsPath = path.resolve(__dirname, "db", "contacts.json");
```
Comment (odykyi) [inline contacts.js]: make it as constant

        this.FILE_CONTACTS_PATH = path.resolve(__dirname, "db", "contacts.json");

File: contacts.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    };
+
+    addContact = async (name, email, phone) => {
+        try {
+            const contactsData = await this.listContacts();
+            const id = contactsData.length ? [...contactsData].pop().id + 1 : 1; //or contactsData.length + 1
+            const newContact = { id, name, email, phone };
+            contactsData.push(newContact);
+            const contactsDataJson = JSON.stringify(contactsData);
+            fs.writeFile(this.fileContactsPath, contactsDataJson);
```
Comment (odykyi) [inline contacts.js]: await

File: contacts.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+            console.log("Error adding new contact", err);
+        }
+    };
+    removeContact = async (contactId) => {
+        try {
+            const contactsData = await this.listContacts();
+            const result = contactsData.filter(
+                (contact) => contact.id !== contactId
+            );
+            fs.writeFile(this.fileContactsPath, JSON.stringify(result));
```
Comment (odykyi) [inline contacts.js]: await

**Review by odykyi — CHANGES_REQUESTED** [review]
comments

**Comment by ekaterina8aleksa** [comment]
bug fixing

File: contacts.js [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    };
+
+    addContact = async (name, email, phone) => {
+        try {
+            const contactsData = await this.listContacts();
+            const id = contactsData.length ? [...contactsData].pop().id + 1 : 1; //or contactsData.length + 1
+            const newContact = { id, name, email, phone };
+            await contactsData.push(newContact);
+            const contactsDataJson = JSON.stringify(contactsData);
+            fs.writeFile(this.FILE_CONTACTS_PATH, contactsDataJson);
```
Comment (odykyi) [inline contacts.js]: add await

File: contacts.js [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+            console.log("Error adding new contact", err);
+        }
+    };
+    removeContact = async (contactId) => {
+        try {
+            const contactsData = await this.listContacts();
+            const result = await contactsData.filter(
+                (contact) => contact.id !== contactId
+            );
+            fs.writeFile(this.FILE_CONTACTS_PATH, JSON.stringify(result));
```
Comment (odykyi) [inline contacts.js]: add await

**Review by odykyi — CHANGES_REQUESTED** [review]
new comments

**Comment by ekaterina8aleksa** [comment]
adding await

**Comment by ekaterina8aleksa** [comment]
hope finish

**Review by odykyi — APPROVED** [review]

**Pull request merged**